submission/file-sources/src/main/kotlin/uk/ac/ebi/io/sources/SubmissionFilesSource.kt

```diff
@@ -23,7 +26,15 @@ internal class SubmissionFilesSource(
         get() = "Previous version files"
 
     override suspend fun getExtFile(path: String, type: String, attributes: List<Attribute>): ExtFile? {
-        return findSubmissionFile(path)?.copyWithAttributes(attributes.map { it.toExtAttribute() })
+        val filePath = expectedPath(path, type)
+        return findSubmissionFile(filePath)?.copyWithAttributes(attributes.map { it.toExtAttribute() })
+    }
+
+    private fun expectedPath(path: String, type: String): String {
+        return when (sub.storageMode) {
+            StorageMode.FIRE -> if (type == FileFields.DIRECTORY_TYPE.value) path.ensureSuffix(".zip") else path
```

Comment:
I think we should leave a TODO here to remove this code once all the submissions have been migrated to FIRE since this looks like a data issue more than a code bug. WDYT?